src/main/java/com/thealgorithms/datastructures/lists/SinglyLinkedList.java

```diff
@@ -119,10 +119,33 @@ public void swapNodes(int valueFirst, int valueSecond) {
     }
 
     /**
-     * Reverse a singly linked list from a given node till the end
+     * Reverse a singly linked list[Iterative] from a given node till the end
      *
      */
-    public Node reverseList(Node node) {
+    public Node reverseListIter(Node node) {
+        Node prev = null;
+        Node curr = node;
+
+        while (curr != null && curr.next != null) {
+            Node next = curr.next;
+            curr.next = prev;
+            prev = curr;
+            curr = next;
+        }
+        // when curr.next==null, the current element is left without pointing it to its prev,so
+        if (curr != null) {
+            curr.next = prev;
+            prev = curr;
+        }
+        // prev will be pointing to the last element in the Linkedlist, it will be the new head of
+        // the reversed linkedlist
+        return prev;
+    }
```

Comment:
How is it different from the method below?

Reply:
There are two methods Iterative and Recursive.
I have added Recursive solutions.
You have mentioned here it's an iterative solution.

Reply:
Extremely Sorry! This is my mistake. Actually, It's a Mistake.
I added the actual reverse code. yes, you can check.